.github/actions/environment_setup/action.yml

```diff
@@ -0,0 +1,17 @@
+---
+name: Enviroment Setup
+description: Setup the enviroment for the other jobs
+
+inputs:
+  python-version:
+    description: 'Version of python to use'
+    required: true
+    default: "3.9"
+
+runs:
+  using: "composite"
+  steps:
+    - name: 🐍 Switch Python ${{ inputs.python-version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ inputs.python-version }}
```

Comment:
This is one case where parameterizing "3.9" creates a ton of syntactic junk. I'm not sure this is a win.

Better might be an encapsulation (refactor) of this whole workflow block
```
    name: ☁️ Pytorch Image Tests [Container]
    runs-on: ubuntu-latest
    steps:
      - name: 🐄 checkout armory full depth with tags for scm
        uses: actions/checkout@v3
        with:
          fetch-depth: 0

      # Get history and tags for SCM versioning to work
      # yamllint disable rule:indentation
      - name: 📩 Checkout the Source
        run: |
          git fetch --prune --unshallow || true
          git fetch --depth=1 origin +refs/tags/*:refs/tags/*
          # Verify .git folder
          if [ ! -d "./.git" ]; then
            echo "Expected ./.git folder to exist"
            exit 1
          fi
          python -m setuptools_scm
          pwd
          git log --oneline --decorate -n 1
      # yamllint enable rule:indentation

      - name: 🐍 Use Python 3.9
        uses: actions/setup-python@v4
        with:
          python-version: 3.9
```

Reply:
That action is not being used as of yet so I just removed the entire directory.